modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -76,39 +81,37 @@ def enum_tree(tree, share, subdir = '')
     write = tree.permissions.write_ea == 1
     skip = false
 
-    if ENUMERABLE_SHARE_TYPES.include? share[:type]
+    if ENUMERABLE_SHARE_TYPES.include?(share[:type])
       msg = share[:type]
-    elsif SKIPPABLE_SHARE_TYPES.include? share[:type]
+    elsif SKIPPABLE_SHARE_TYPES.include?(share[:type])
       msg = share[:type]
       skip = true
     else
       msg = "Unhandled Device Type (#{share[:type]})"
       skip = true
     end
 
-    print_status "Skipping share #{share[:name].strip} as it is of type #{share[:type]}" if skip
+    print_status("Skipping share #{share[:name].strip} as it is of type #{share[:type]}") if skip
     return read, write, msg, nil if skip
 
     # Create list after possibly skipping a share we wouldn't be able to access.
     begin
       list = tree.list(directory: subdir)
     rescue RubySMB::Error::UnexpectedStatusCode => e
-      print_error e.to_s
-      return read, write, msg, nil
+      vprint_error("Error when trying to list tree contents in #{share[:name]}\\#{subdir} - #{e.status_code.name}")
     end
 
     rfd = []
-    list.entries.each do |file|
-      file_name = file.file_name.strip.encode('UTF-8')
-      next if file_name == '.' || file_name == '..'
+    if list
```

Comment:
This doesn't make sense to me:

```
 ~/git/metasploit-framework │ upstream-master:master ?35  irb         ✔ │ 3s │ 3.0.2 Ruby 
3.0.2 :001 > if []
3.0.2 :002 >   print "A"
3.0.2 :003 > end
A => nil 
3.0.2 :004 > if false
3.0.2 :005 >   print "B"
3.0.2 :006 > end
 => nil 
3.0.2 :007 > if nil
3.0.2 :008 >   print "C"
3.0.2 :009 > end
 => nil 
3.0.2 :010 >  
```

A suggestion would be `unless list.nil? || list.empty?`

Reply:
Resolved in 147ca34308f76997ca2f58b8d10abeb9e9372182